Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoDecoderFactory.cpp

```diff
@@ -212,6 +212,11 @@ class GStreamerWebRTCVideoDecoder : public webrtc::VideoDecoder {
             return WEBRTC_VIDEO_CODEC_UNINITIALIZED;
         }
 
+        if (inputImage._encodedWidth > 0 && inputImage._encodedHeight > 0) {
```

Comment:
Would it make sense to keep track of the input image dimensions and update the caps only when those dimensions change?

With the current version of this patch I'm a bit worried with the extra allocations.

Reply:
Yes, of course. I've updated the branch

Reply:
Thanks! I can take care of upstreaming and backporting of this PR.